modules/e2ee/Context.js

```diff
@@ -268,121 +171,38 @@ export class Context {
      * @param {RTCEncodedVideoFrame|RTCEncodedAudioFrame} encodedFrame - Encoded video frame.
      * @param {TransformStreamDefaultController} controller - TransportStreamController.
      */
-    async decodeFunction(encodedFrame, controller) {
+    decodeFunction(encodedFrame, controller) {
         const data = new Uint8Array(encodedFrame.data);
-        const keyIndex = data[encodedFrame.data.byteLength - 1] & 0xf; // lower four bits.
-
-        if (this._cryptoKeyRing[this._currentKeyIndex] && this._cryptoKeyRing[keyIndex]) {
-            const counterLength = 1 + ((data[encodedFrame.data.byteLength - 1] >> 4) & 0x7);
-            const signatureLength = data[encodedFrame.data.byteLength - 1] & 0x80
-                ? this._signatureOptions.byteLength : 0;
-            const frameHeader = new Uint8Array(encodedFrame.data, 0, UNENCRYPTED_BYTES[encodedFrame.type]);
-
-            // Extract the truncated authentication tag.
-            const authTagOffset = encodedFrame.data.byteLength - (DIGEST_LENGTH[encodedFrame.type]
-                + counterLength + signatureLength + 1);
-            const authTag = encodedFrame.data.slice(authTagOffset, authTagOffset
-                + DIGEST_LENGTH[encodedFrame.type]);
-
-            // Verify the long-term signature of the authentication tag.
-            if (signatureLength) {
-                if (this._signatureKey) {
-                    const signature = data.subarray(data.byteLength - signatureLength - 1, data.byteLength - 1);
-                    const validSignature = await crypto.subtle.verify(this._signatureOptions,
-                            this._signatureKey, signature, authTag);
-
-                    if (!validSignature) {
-                        // TODO: surface this to the app. We are encrypted but validation failed.
-                        console.error('Long-term signature mismatch (or no signature key)');
-
-                        return;
-                    }
-
-                    // TODO: surface this to the app. We are now encrypted and verified.
-                } else {
-                    // TODO: surface this to the app. We are now encrypted but can not verify.
-                }
-
-                // Then set signature bytes to 0.
-                data.set(new Uint8Array(signatureLength), encodedFrame.data.byteLength - (signatureLength + 1));
-            }
-
-            // Set authentication tag bytes to 0.
-            data.set(new Uint8Array(DIGEST_LENGTH[encodedFrame.type]), encodedFrame.data.byteLength
-                - (DIGEST_LENGTH[encodedFrame.type] + counterLength + signatureLength + 1));
-
-            // Do truncated hash comparison of the authentication tag.
-            // If the hash does not match we might have to advance the ratchet a limited number
```

Comment:
We still want to do the ratcheting, but now when decryption fails instead, assuming we have the key.